components/bl-dashlet/src/helpers.js

```diff
@@ -124,3 +124,83 @@ export const getPosition = (ref, coords) => {
     ),
   };
 };
+
+export const useContextMenuPositionHandler = (contextMenuRef, isMenuOpen) => {
+  const [sides, setSides] = useState({ left: false, top: false });
+  const [isChecked, setIsChecked] = useState(false);
+
+  useEffect(() => {
+    if (contextMenuRef.current && isMenuOpen) {
+      const { bottom, right } = contextMenuRef.current.getBoundingClientRect();
+
+      if (right > window.innerWidth) {
+        sides.left = true;
+      }
+
+      if (bottom > window.innerHeight) {
+        sides.top = true;
```

Comment:
It will not trigger rerender, as you didn't call setSides. Also I'm not clearly understand why we need useState and useEffect if it could be useMemo.

And the last. Does it works when page resized?

Reply:
I tried to use useMemo, but when I open the menu contextMenuRef is empty, when close menu contextMenuRef has the element

```
export const useContextMenuPositionHandler = (contextMenuRef, isMenuOpen) => useMemo(() => {
  const sides = { left: false, top: false };
  let isChecked;

  if (contextMenuRef.current && isMenuOpen) {
    const { bottom, right } = contextMenuRef.current.getBoundingClientRect();

    if (right > window.innerWidth) {
      sides.left = true;
    }

    if (bottom > window.innerHeight) {
      sides.top = true;
    }

    isChecked = true;
  } else {
    sides.left = false;
    sides.top = false;
    isChecked = false;
  }

  return { sides, isChecked };
}, [contextMenuRef, isMenuOpen]);
```

Reply:
How offen `contextMenuRef` will be changed? Think about it and answer me in slack.